### PR TITLE
Dev build system 0.0 to dev

### PR DESCRIPTION
**Pull request by bfilonenko**

### CODE REVIEW

**Review by SymphonyOfTranquility — APPROVED** [review]

File: Assets/Scripts/Builder/Builder.cs [hunk 1]
```diff
@@ -53,17 +55,26 @@ private void UpdateCurrentBuilding()
...
-                currentBuildingPrefab,
-                currentBuilding.transform.position - Vector3.up * preViewOffset,
-                Quaternion.identity);
+            Vector3 position = currentBuilding.transform.position - Vector3.up * preViewOffset;
+            GameObject building = Instantiate(currentBuildingPrefab, position, Quaternion.identity);
+
+            if (isCurrentBuildResourceExtractor)
+            {
+                ResourceDeposit resourceDeposit = worldGenerator.GetResourceDeposit(position);
+                Debug.Assert(resourceDeposit, "Resource Deposit is null");
```
Comment (Afohtim) [inline Assets/Scripts/Builder/Builder.cs:73]: Why is this should be null?

Comment (bfilonenko) [reply Assets/Scripts/Builder/Builder.cs:73]: If hex cell on this position doesn't contain resource, ```worldGenerator.GetResourceDeposit``` function will return null.

Comment (Afohtim) [reply Assets/Scripts/Builder/Builder.cs:73]: But it asserts is should be null all the time

Comment (bfilonenko) [reply Assets/Scripts/Builder/Builder.cs:73]: Why?

**Review by Afohtim — APPROVED** [review]

**Pull request merged**